WeakAuras/BossMods.lua

```diff
@@ -0,0 +1,1598 @@
+      and bar.expirationTime + extendTimer > GetTime()
+      then
+        bestMatch = bar
+      end
+    end
+    return bestMatch
+  end,
+
+  RecheckTimers = function(self)
+    local now = GetTime()
+    self.nextExpire = nil
+    for timerId, bar in pairs(self.bars) do
+      if not bar.paused then
+        if bar.expirationTime < now then
+          self.bars[timerId] = nil
+          WeakAuras.ScanEvents("DBM_TimerStop", timerId)
+          if self.isGeneric then
+            WeakAuras.ScanEvents("BossMod_TimerStop", timerId)
+          end
+        elseif self.nextExpire == nil then
+          self.nextExpire = bar.expirationTime
+        elseif bar.expirationTime < self.nextExpire then
+          self.nextExpire = bar.expirationTime
+        end
+      end
+    end
+    if self.nextExpire then
+      self.recheckTimer = timer:ScheduleTimerFixed(self.RecheckTimers, self.nextExpire - now, self)
+    end
+  end,
+
+  EventCallback = function(self, event, ...)
+    if event == "DBM_Announce" then
+      local spellId = select(4, ...)
+      WeakAuras.ScanEvents("DBM_Announce", spellId, ...)
+      if self.isGeneric then
+        WeakAuras.ScanEvents("BossMod_Announce", spellId, ...)
+      end
+    elseif event == "DBM_TimerStart" then
+      local timerId, msg, duration, icon, timerType, spellId, dbmType, _, _, _, _, _, timerCount = ...
+      local now = GetTime()
+      local expirationTime = now + duration
+      self.bars[timerId] = self.bars[timerId] or {}
+      local bar = self.bars[timerId]
+      bar.message = msg
+      bar.expirationTime = expirationTime
+      bar.duration = duration
+      bar.icon = icon
+      bar.timerType = timerType
+      bar.spellId = tostring(spellId)
+      bar.count = timerCount and tostring(timerCount) or "0"
+      bar.dbmType = dbmType
+
+      local barOptions = DBT.Options or DBM.Bars.options
+      local r, g, b = 0, 0, 0
+      if dbmType == 1 then
+        r, g, b = barOptions.StartColorAR, barOptions.StartColorAG, barOptions.StartColorAB
+      elseif dbmType == 2 then
+        r, g, b = barOptions.StartColorAER, barOptions.StartColorAEG, barOptions.StartColorAEB
+      elseif dbmType == 3 then
+        r, g, b = barOptions.StartColorDR, barOptions.StartColorDG, barOptions.StartColorDB
+      elseif dbmType == 4 then
+        r, g, b = barOptions.StartColorIR, barOptions.StartColorIG, barOptions.StartColorIB
+      elseif dbmType == 5 then
+        r, g, b = barOptions.StartColorRR, barOptions.StartColorRG, barOptions.StartColorRB
+      elseif dbmType == 6 then
+        r, g, b = barOptions.StartColorPR, barOptions.StartColorPG, barOptions.StartColorPB
+      elseif dbmType == 7 then
+        r, g, b = barOptions.StartColorUIR, barOptions.StartColorUIG, barOptions.StartColorUIB
+      else
+        r, g, b = barOptions.StartColorR, barOptions.StartColorG, barOptions.StartColorB
+      end
+      bar.dbmColor = {r, g, b}
```

Comment:
We can likely expose the colorVars table in DBT.

Though also, you should probably be doing `barOptions['StartColor' .. colorVariables[dbmType] .. 'G']` etc., rather than a nasty switch mess?

Reply:
https://github.com/DeadlyBossMods/DBM-Unified/commit/08166b43c2f031de988597ffaab21c5fd74f3b75 Should help with this =]

Reply:
I'd like to add that these are NOT option types, just literally color assignments, type is an entirely differet arg in the actual callback, these should never be used to assign type to a weak aura, ever. In fact I just refactored them to do away with them being types in the GUI and now they're just color 1 through color 8 cause users had same confusion. thinking it set a timer type when all it sets is bar color. The types were just what that color was assigned to by default and nothing more. In fact, for this PR, i don't even see how they're relevant at all since what color DBM makes a bar isn't used in these triggers and as far as I've seen type is also omitted. if type is added though then refer to https://github.com/WeakAuras/WeakAuras2/pull/4537#issuecomment-1642607655

Reply:
The 'dbmType' is a misnaming on our end, it is what is called colorType in the dbm sources, so yes that's about the color not about the timer type.